packages/dcos-integration-test/extra/test_metrics.py

```diff
@@ -100,22 +100,53 @@ def test_metrics_containers(dcos_api_session):
     def test_containers(app_endpoints):
         for agent in app_endpoints:
             response = dcos_api_session.metrics.get('containers', node=agent.host)
+            assert response.status_code == 200, "expected 200 status code from /containers/, got {}".format(
+                response.status_code)
+
             if len(response.json()) > 0:
                 for c in response.json():
-
                     # Test that /containers/<id> responds with expected data
-                    container_response = dcos_api_session.metrics.get('containers/{}'.format(c), node=agent.host)
-                    if container_response.status_code == 200 and 'executor_id' in container_response.json():
-                        # Executor ID value is "executor_id": "statsd-emitter.a094eed0-b017-11e6-a972-b2bcad3866cb"
-                        assert 'statsd-emitter' in container_response.json()['executor_id'].split('.'), 'statsd-emitter'
-                        ' was not found running on any slaves.'
+                    container_id_path = 'containers/{}'.format(c)
+                    container_response = dcos_api_session.metrics.get(container_id_path, node=agent.host)
+
+                    # /contiainers/<container_id> should always respond succesfully
+                    assert container_response.status_code == 200, 'expected 200 response from {}, got {}'.format(
+                        'containers/{}'.format(c),
```

Comment:
If you want to keep the reason string you should probably use `container_id_path` here.